sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -2132,7 +2133,7 @@ export class SearchIndexClient {
     getAlias(aliasName: string, options?: GetAliasOptions): Promise<SearchIndexAlias>;
     getIndex(indexName: string, options?: GetIndexOptions): Promise<SearchIndex>;
     getIndexStatistics(indexName: string, options?: GetIndexStatisticsOptions): Promise<SearchIndexStatistics>;
-    getSearchClient<T>(indexName: string, options?: SearchClientOptions): SearchClient<T>;
+    getSearchClient<Model extends object>(indexName: string, options?: SearchClientOptions): SearchClient<Model>;
```

Comment:
This has similar semantics to the `SearchClient` constructor; documents are coerced/deserialized into a `Model`. Type hints are only available when `Model` is passed explicitly to the constructor.

Reply:
I'm realizing there's a bit of friction in users who define a `SearchIndex` and create their index, as they _also_ need to define the corresponding TS type to pass as the generic type of `getSearchClient` or `new SearchClient`. It may be worth considering allowing the user to pass `typeof index` as the generic type of these functions by parsing the model out.